mono/mini/mini-amd64.h

```diff
@@ -584,8 +584,10 @@ mono_arch_unwindinfo_free_unwind_info (PUNWIND_INFO unwind_info);
 guint
 mono_arch_unwindinfo_init_method_unwind_info (gpointer cfg);
 
+#ifdef MONO_ARCH_HAVE_UNWIND_TABLE
```

Comment:
Is this #ifdef needed? The define is set above and all of the methods in this block are enabled so nothing special about this method.

Reply:
Maybe not.

Reply:
LPDWORD is a leftover from 16bit systems with "long" pointers and has no place in today's code.

Reply:
I think you are right. Let me check and maybe remove it. You just can't use that type in portable code.

Reply:
You also use it right below.

Reply:
I see. "Right below" after the endifs, in which it is defined.

Reply:
I was also surprised you typedef UNWIND_INFO and UNWIND_CODE yourself vs. from windows.h. windows.h does have a big problem, granted -- it doesn't allow for cross-compilation or cross-dumping (link /dump). The types in windows.h should be AMD64_UNWIND_INFO and then conditionally typedefed to "native" UNWIND_INFO. Many folks are forced to duplicate them because of this, alas, i.e. all cross-compilers and cross-linkers, i.e. all compilers and linkers, except for the occasional JIT-only.

Reply:
Lately people are even advising against the "P" and favoring DWORD* over PDWORD. The "P" is also kind of a holdover, from NT style and I suspect it dates to more verbose language like Pascal or Modula or PL/1 where the "P" was really a tremendous shortening over like REF DWORD or DWORD POINTER; TYPE PDWORD = REF DWORD or such. But in C it is just trading one character for one character, two tokens for one token.